declcache.go

```diff
@@ -224,9 +225,23 @@ func build_package(p *build.Package) error {
 		log.Printf("package object: %s", p.PkgObj)
 		log.Printf("package source dir: %s", p.Dir)
 		log.Printf("package source files: %v", p.GoFiles)
+		log.Printf("GOPATH: %v", g_daemon.context.GOPATH)
+		log.Printf("GOROOT: %v", g_daemon.context.GOROOT)
 	}
+	env := os.Environ()
+	for i, v := range env {
+		if strings.HasPrefix(v, "GOPATH=") {
+			env[i] = "GOPATH=" + g_daemon.context.GOPATH
+		} else if strings.HasPrefix(v, "GOROOT=") {
```

Comment:
Thanks. I've never been clear why `Environ()` returns a `[]string` and not a `map[string]string` (given `Getenv(k string)` returns a `string` and `Setenv` takes a single val `string`). Your code clearly assumes the `[]string` will be unique by key... which seems logical to me given my experience with `bash` et al, but do we need to defensively code for the case the slice is not unique by key?